In [ ]:
# Import libraries
import pandas as pd
import numpy as np
import ast
import re
import math

!! pip install geopy
from geopy.geocoders import Nominatim
# libraries for displaying images
from IPython.display import Image 
from IPython.core.display import HTML 

import matplotlib.pyplot as plt
import seaborn as sns


In [5]:
df = pd.read_csv(r"C:\Users\dell\Desktop\Data Analyst Project\zomato2\zomato.csv", encoding='utf-8')

Lets drop duplicates based on the value from two columns address and name

In [6]:
df = df.drop_duplicates(subset=['address', 'name']).reset_index().drop('index', axis=1)
df.head()

url  \
0  https://www.zomato.com/bangalore/jalsa-banasha...   
1  https://www.zomato.com/bangalore/spice-elephan...   
2  https://www.zomato.com/SanchurroBangalore?cont...   
3  https://www.zomato.com/bangalore/addhuri-udupi...   
4  https://www.zomato.com/bangalore/grand-village...   

                                             address                   name  \
0  942, 21st Main Road, 2nd Stage, Banashankari, ...                  Jalsa   
1  2nd Floor, 80 Feet Road, Near Big Bazaar, 6th ...         Spice Elephant   
2  1112, Next to KIMS Medical College, 17th Cross...        San Churro Cafe   
3  1st Floor, Annakuteera, 3rd Stage, Banashankar...  Addhuri Udupi Bhojana   
4  10, 3rd Floor, Lakshmi Associates, Gandhi Baza...          Grand Village   

  online_order book_table   rate  votes                             phone  \
0          Yes        Yes  4.1/5    775    080 42297555\r\n+91 9743772233   
1          Yes         No  4.1/5    787                      080 41714161   
2          Yes         No  3.8/5    918                    +91 9663487993   
3           No         No  3.7/5     88                    +91 9620009302   
4           No         No  3.8/5    166  +91 8026612447\r\n+91 9901210005   

       location            rest_type  \
0  Banashankari        Casual Dining   
1  Banashankari        Casual Dining   
2  Banashankari  Cafe, Casual Dining   
3  Banashankari          Quick Bites   
4  Basavanagudi        Casual Dining   

                                          dish_liked  \
0  Pasta, Lunch Buffet, Masala Papad, Paneer Laja...   
1  Momos, Lunch Buffet, Chocolate Nirvana, Thai G...   
2  Churros, Cannelloni, Minestrone Soup, Hot Choc...   
3                                        Masala Dosa   
4                                Panipuri, Gol Gappe   

                         cuisines approx_cost(for two people)  \
0  North Indian, Mughlai, Chinese                         800   
1     Chinese, North Indian, Thai                         800   
2          Cafe, Mexican, Italian                         800   
3      South Indian, North Indian                         300   
4        North Indian, Rajasthani                         600   

                                        reviews_list menu_item  \
0  [('Rated 4.0', 'RATED\n  A beautiful place to ...        []   
1  [('Rated 4.0', 'RATED\n  Had been here for din...        []   
2  [('Rated 3.0', "RATED\n  Ambience is not that ...        []   
3  [('Rated 4.0', "RATED\n  Great food and proper...        []   
4  [('Rated 4.0', 'RATED\n  Very good restaurant ...        []   

  listed_in(type) listed_in(city)  
0          Buffet    Banashankari  
1          Buffet    Banashankari  
2          Buffet    Banashankari  
3          Buffet    Banashankari  
4          Buffet    Banashankari

First let us make food corpus for our data

Since the column reviews_list, menu_item are in the form of list of tuples but as we read it from file we need to explicitly convert it to list

In [7]:
df['reviews_list'] =  df['reviews_list'].apply(ast.literal_eval)
df['menu_item'] =  df['menu_item'].apply(ast.literal_eval)

In [9]:
li = []
def func(x):
    global li
    li = li + x
a = df[df['menu_item']!="[]"]['menu_item'].apply(func)

In [10]:
# convert all the menu to lower case letter and storing in the variable food_corpus
food_corpus = list(map(str.lower, set(li)))

In [ ]:
## food_corpus

On analysing the corpus it was found that some food contains quantity written beside it inside the brackets.
Let us remove those brackets.

In [11]:
menu_corpus = []
for x in food_corpus:
    menu_corpus.append(re.sub("\[.*\]|\(.*\)", "", x).strip())

In [12]:
len(menu_corpus)

71424

Very well, so we have 71,424 food items in our corpus. But there is no data without noise, let us remove noise from it.

Lets analyse the data, for the words that frequently occur and are not a food item.

In [15]:
ind = df[df['reviews_list']!="[]"].index
for i in ind:
    liked = set()
    for review in df.loc[i,'reviews_list']:
        if(float(review[0].replace('Rated',''))>=3):
            for menu in menu_corpus:
                if(menu in review[1]):
                    liked.add(menu)
    print(liked)
                

{'butter', 'roti', 'paratha.', 'fried rice', 'bread', 'butter roti', 'chilli paratha', 'masala papad', 'kulcha', 'palak', 'rice', 'paratha', 'soya chaap', 'papad', 'fried', 'baby corn', 'poi'}
{'undo', 'roti', 'sweet', 'chicken', 'paneer', 'egg', 'spicy', 'sweet and spicy', 'veggie', 'burnt garlic fried rice', 'garlic fried rice', 'veg', 'rice', 'chicken biryan', 'fried', 'mix veg', 'uttar dakshin', 'paneer kurchan', 'fried rice', 'lunch', 'chicken biryani'}
{'coconut milk', 'lasagne', 'nutella', 'sweet', 'egg', 'chocolate', 'hot chocolate', 'choco', 'spicy', 'disco', 'milk', 'sicilia pizza', 'hot choco', 'coffee', 'pizza', 'veggie', 'veg', 'churros', 'rice', 'cake', 'nachos', 'special', 'smile', 'pink pasta', 'chocolate sauce', 'veggies', 'shakes', 'pasta', 'water', 'poi'}
{'sambar', 'full meal', 'holige', 'full meals', 'sweet', 'pulav', 'rice bath', 'apple', 'pulka', 'limited meal', 'rasam', 'veg rice', 'payasam', 'veg', 'meal', 'rice with sambar', 'rice', 'dosa', 'pickle', 'ghee', '

{'panneer butter masala', 'sev', 'grill', 'latte', 'chapati', 'butter chicken curry', 'kabab', 'chicken gravy', 'chicken kadai', 'tandoori chicken', 'grilled chicken', 'prawns', 'paneer butter masala', 'butter masala', 'dal', 'lime soda', 'shawarma', 'rice', 'alfaham', 'shakes', 'butter', 'roti', 'chai', 'grape', 'coin parota', 'paneer butter', 'fish', 'chicken', 'lollipop', 'anjal', 'parota', 'spicy', 'roll', 'pepper chicken', 'veg lovers', 'chili chicken', 'paneer butter masala.', 'ghee', 'anjal tawa', 'lunch', 'grill chicken', 'kadai', 'fresh lime', 'chicken wings', 'special chicken', 'noodles', 'naan', 'fresh lime soda', 'chocolate', 'prawns tawa fry', 'choco', 'al faham', 'tawa fry', 'chilly chicken', 'veg', 'grape juice', 'chicken biryan', 'special', 'butter chicken', 'ghee rice', 'thali', 'chicken curry', 'kadai chicken', 'soda', 'prawns tawa', 'paneer', 'tandoori roti', 'lime', 'alfaham chicken', 'chicken skewer', 'pulpy grape juice', 'tandoori', 'mutton biryani', 'meal', 'dosa

{'butter', 'onion raitha', 'chicken', 'paneer', 'roasted', 'egg chilli', 'egg', 'spicy', 'poratta', 'mushroom fry', 'mint cooler', 'raitha', 'rice', 'onion', 'rabadi', 'fried', 'special', 'mushroom', 'tea', 'spice', 'poi'}
{'roll', 'chicken rolls', 'chicken', 'chicken roll'}
{'butter', 'veg red sauce pasta', 'roti', 'butter naan', 'khichdi', 'seu', 'chicken tikka', 'red sauce pasta', 'baby corn chilly', 'chicken noodles.', 'noodles', 'chicken', 'butter naan.', 'naan', 'paneer tikka', 'paneer', 'cold coffe', 'roomali roti', 'spicy', 'crispy babycorn', 'garlic butter naan', 'baby corn', 'cold coffee', 'dal', 'coffee', 'schezwan noodles', 'veg', 'bread', 'meal', 'rice', 'chicken noodles', 'garlic bread', 'dal khichdi', 'jamun', 'fried', 'special', 'veg pasta', 'non-veg', 'schezwan noodle', 'chicken tikka masala', 'curd rice', 'curd', 'white sauce pasta', 'fried rice', 'chicken noodle', 'kit kat', 'naan.', 'baby corn chil', 'kit kat shake', 'churi', 'pasta', 'water', 'poi', 'packing'}
{'ro

{'popcorn', 'chicken', 'chicken pop', 'chicken popcorn'}
set()
{'vanilla', 'kulfi', 'sweet', 'blueberry', 'chocolate', 'choco', 'roll', 'milk', 'wafer', 'coffee', 'rice', 'kesar kulfi', 'special', 'blueberry ice cream', 'lychee', 'chocolate fudge', 'ice cream', 'overload', 'banana', 'vanilla ice cream', 'fudge', 'chocolate sauce'}
{'butter', 'butter naan', 'biriyani rice.', 'noodles', 'chicken', 'naan', 'panner butter', 'prawns', 'prawns fried rice', 'chicken fried rice', 'butter masala', 'panner butter masala', 'rice', 'chicken noodles', 'fried', 'biriyani rice', 'fried rice', 'chicken noodle', 'prawns fried rice.'}
{'chai'}
set()
{'chicken rolls', 'chicken tikka', 'sweet', 'chicken', 'crispy chicken', 'chicken kabab.', 'kebab', 'kabab', 'roll', 'spicy', 'chicken gravy', 'tandoori chicken', 'shadi ka biryani', 'packing', 'tandoori kebab', 'tandoori', 'crispy chicken biryani', 'veg lovers', 'malai kofta', 'mutton biryani', 'lemon chicken', 'veg', 'meal', 'chicken roll', 'rice', 'onion'

{'chai', 'samosa', 'kulfi', 'sweet', 'puri', 'dahi puri', 'lime', 'milk', 'moti chur laddu', 'kachori', 'dahi', 'cap', 'special', 'laddu', 'badam milk', 'curd', 'badam', 'legendary', 'rasmalai', 'idli'}
{'cheese', 'mixed', 'mayo', 'spicy', 'baby corn', 'mixed sauce pasta', 'poi', 'mayonnaise', 'bread', 'rice', 'garlic bread', 'fried', 'smile', 'mushroom', 'cheese garlic bread', 'fried rice', 'mushroom pepper', 'fries', 'pasta', 'triple schezwan fried rice', 'schezwan fried rice'}
{'butter', 'akki rotti', 'butter fruit juice', 'rice bath', 'fruit juice', 'filter coffee', 'set dosa', 'lime', 'rava dosa', 'milk', 'pulao', 'sandwich', 'coffee', 'veg', 'butterscotch', 'butterscotch milkshake', 'rice', 'dosa', 'onion', 'butter fruit', 'capsicum', 'milkshake', 'cap', 'special', 'masala dosa', 'onion dosa', 'pongal', 'rawa dosa', 'poi'}
{'pizza', 'veg', 'burger', 'cheese', 'onion', 'capsicum', 'paneer', 'cap', 'sandwich'}
{'kulfi', 'sweet', 'lassi', 'hot jamun', 'latte', 'sarson ka saag', 'sal

{'smile', 'vadapav', 'rice', 'pav', 'special', 'vada', 'sandwich', 'packing'}
{'spicy', 'biryani rice', 'rice', 'chicken manchurian', 'chicken manchuri', 'chicken', 'churi', 'egg', 'packing'}
{'coffee', 'fried rice', 'lunch', 'fried', 'meal', 'rice', 'onion', 'adas', 'noodles', 'churi', 'special', 'vada'}
{'undo', 'sev', 'sweet', 'crab fry', 'latte', 'hot and sour soup', 'chapati', 'kabab', 'chicken fry', 'chicken gravy', 'rasam', 'prawns', 'chicken hot and sour', 'dal', 'lemon chicken', 'crab', 'rice', 'gongura mutton', 'fry piece biryani', 'chicken lollipop.', 'non-veg', 'andhra meals', 'ghee rice.', 'sambar', 'chicken hot', 'fried chicken', 'chicken hot and sour soup', 'lollipop', 'chicken', 'fish', 'kebab', 'veg thali', 'limited meal', 'spicy', 'carrier meals', 'special kabab', 'chilli chicken', 'mutton fry', 'fish curry', 'golden prawns', 'ghee', 'kheema', 'chicken majestic', 'lunch', 'spice', 'jeera rice', 'veg dum biryani', 'veg fried rice', 'special chicken', 'fish curry.', 'hu

{'rice bath', 'masala dosa', 'rice', 'dosa'}
{'butter', 'roti', 'saga', 'north indian thali', 'carrot', 'halwa', 'north indian meals', 'paneer butter', 'pav', 'pav bhaji', 'paneer chilly', 'puri', 'paneer', 'bonda soup', 'kesari', 'paper masala', 'mini meals', 'rotis', 'indian meals', 'milk', 'paneer butter masala', 'baby corn', 'butter masala', 'bhaji', 'sandwich', 'bondas', 'coffee', 'gobi manchuri', 'carrot halwa', 'mini meal', 'meal', 'rice', 'dosa', 'paneer butter masala.', 'milkshake', 'bonda', 'special', 'fried', 'masala dosa', 'thali', 'masala puri', 'fried rice', 'paper masala dosa', 'lunch', 'ice cream', 'north indian meal', 'kharabath', 'idli', 'indian meal', 'churi', 'shakes', 'bhajji', 'water', 'poi'}
{'kotte kadubu', 'thumbs up', 'wheat pongal', 'pongal'}
{'roti', 'prawn ghee roast', 'appam', 'chicken gravy', 'rice', 'chicken', 'ghee', 'ghee roast', 'kori roti'}
{'fried rice', 'rice', 'schezwan fried rice', 'fried'}
{'mojito', 'cheese', 'cheese ball', 'lassi', 'fries', 'c

{'cheese fries', 'soda', 'tender coconut ice cream', 'sev', 'cheese', 'birthday treat', 'fish', 'chicken', 'apple', 'tender coconut', 'egg', 'egg roll', 'kebab', 'coconut ice cream', 'mayo', 'kabab', 'roll', 'spicy', 'milk', 'mango shake', 'fish meals', 'jackfruit ice cream', 'sandwich', 'jackfruit thick shake', 'meal', 'rice', 'milkshake', 'special', 'kheema', 'jackfruit', 'ice cream', 'burger', 'mango', 'fish meal', 'churi', 'fries', 'shakes', 'water', 'poi'}
{'butter', 'tomato shorba', 'carrot', 'halwa', 'cheese', 'naan', 'roasted', 'egg', 'kebab', 'roll', 'lime', 'papad', 'veggie', 'veg', 'bread', 'veggie roll', 'palak', 'rice', 'garlic bread', 'fried', 'special', 'smile', 'fried rice', 'lunch', 'ice cream', 'last night', 'mast', 'water'}
{'red velvet', 'berry shake', 'chocolate', 'kiwi', 'choco', 'red velvet shake', 'disco', 'spinach corn sandwich', 'corn sandwich', 'french fries', 'milk', 'sandwich', 'coffee', 'veg', 'rice', 'strawberry shake', 'smoothie', 'smoothies', 'milkshake

{'butter', 'roti', 'fanta', 'kadhai paneer', 'chai', 'rajma masala', 'sev', 'birthday treat', 'lassi', 'aloo paratha', 'sabji', 'paneer', '2 paratha', 'spicy', 'hulk', 'combo meal', 'disco', 'butter masala', 'rajma', 'paratha.', 'veg', 'meal', 'rice', 'onion', 'phulka', 'tuna', 'pickle', 'special', 'poha', 'ajwain paratha', 'thali', 'mushroom', 'lunch', 'out of the blue', 'paratha', 'missi roti', 'churi', 'poi'}
{'roll', 'burger', 'rice', 'fried momo', 'fried momos', 'paneer', 'hot dog', 'momos', 'fried'}
{'fanta', 'vanilla', 'orange chocolate', 'chocolate', 'orange', 'choco', 'brain', 'tea cake', 'hazel', 'cake', 'eclair', 'tac', 'tea', 'rose', 'passion fruit', 'almond', 'strawberry', 'hazelnut', 'pista', 'poi'}
{'sweet', 'gulab jamun', 'jamun'}
{'roti', 'chai', 'sweet', 'noodles', 'chicken', 'egg', 'kebab', 'triple fried rice', 'veg noodles', 'chicken dum biryani', 'spicy', 'roll', 'veg noodle', 'rotis', 'chicken manchurian', 'mushroom fried rice', 'tandoori chicken', 'chicken manchu

{'butter', 'fruit lassi', 'dal makhni', 'do pyaza', 'cheese', 'pakoda', 'cheese ball', 'lassi', 'dry fruit lassi', 'sweet', 'fries', 'cheesy', 'cheese overload', 'paneer', 'egg', 'cutlet', 'shrikhand', 'paneer lababdar', 'gobi aloo', 'cheese paratha', 'spicy', 'hulk', 'cheese balls', 'bharta', 'french fries', 'papad', 'pulao', 'dal', 'veggie', 'matar pulao', 'paratha.', 'pulkha', 'veg', 'fulka', 'veg cutlet', 'meal', 'rice', 'cheese burst paratha', 'onion', 'cutlets', 'batura', 'dal makhani', 'fruits lassi', 'dry fruits lassi', 'veg kofta', 'aloo gobhi', 'masala fries', 'special', 'fried', 'thali', 'ice cream', 'parata', 'dry fruit', 'overload', 'paratha', 'spicy fries', 'veg cutlets', 'paneer paratha', 'veggies', 'mop', 'water', 'poi', 'dry fruits'}
{'rabadi'}
{'curd', 'veggie', 'curd rice', 'veg', 'mixture', 'meal', 'rice', 'veggies', 'egg'}
{'cheese', 'mosambi', 'pav', 'noodles', 'paneer', 'oreo milk shake', 'oreo', 'mac n cheese', 'roll', 'paneer roll', 'milk', 'vada', 'mosambi jui

{'gobi manchurian', 'watermelon', 'churi', 'noodles', 'gobi manchuria', 'chocolate', 'fresh juice', 'milkshake with ice cream', 'choco', 'potato wedges', 'strawberry milkshake', 'strawberry milkshake with ice cream', 'chocolate milk', 'milk', 'sandwich', 'pizza', 'gobi manchuri', 'rice', 'milkshake', 'fried', 'fried rice', 'ice cream', 'burger', 'chocolate milk shake', 'wedges', 'strawberry', 'fries', 'shakes', 'pasta', 'water', 'poi'}
{'pizza', 'ice cream', 'roll', 'rice', 'fruit juice', 'sandwich'}
set()
{'tornado', 'choco', 'ice cream', 'milk shakes', 'chocolate milk shake', 'rice', 'sweet', 'chocolate milk', 'milk', 'strawberry', 'shakes', 'chocolate', 'poi'}
set()
set()
{'dhokla', 'badam milk', 'non-veg', 'tea', 'badam', 'kalakan', 'veg', 'sweet', 'paratha', 'milk', 'kalakand', 'special'}
{'smile', 'choco', 'ice cream', 'spice', 'rice', 'cake', 'chocolate ice cream', 'chocolate'}
{'butter', 'bun', 'pundi', 'butter murg', 'rice', 'capsicum puff', 'honey cake', 'capsicum', 'cake', '

{'roti', 'saga', 'paneer ghee roast', 'masala papad', 'kulcha', 'fish', 'puri', 'paneer tikka', 'paneer', 'egg', 'hot and sour soup', 'ghee roast', 'capsicum masala', 'spicy', 'rotis', 'corn capsicum masala', 'corn and capsicum masala', 'papad', 'tawa fry', 'dal', 'coffee', 'veggie', 'veg', 'bread', 'meal', 'dal tadka', 'rice', 'dosa', 'onion', 'palak', 'tomato soup', 'capsicum', 'hot and sour', 'cap', 'ghee', 'special', 'fried', 'bun', 'masala dosa', 'curd', 'curd rice', 'palak soup', 'fried rice', 'lunch', 'kharabath', 'thumbs up', 'churi', 'veggies', 'poi'}
set()
{'pineapple', 'kesari', 'coffee', 'legendary', 'carrot halwa', 'kesaribath', 'carrot', 'kesari bath', 'halwa', 'rice', 'dosa', 'khali dosa', 'benne dose', 'peni', 'apple', 'water'}
{'meal'}
{'veg', 'fried chicken', 'chicken', 'egg', 'fried'}
{'chicken curry', 'chicken biryani', 'chicken', 'chicken biryan', 'mudde', 'chilli chicken'}
{'sambar', 'fanta', 'rava idli', 'special meal', 'white rice', 'kesari', 'pulao', 'vada', 'c

{'roti', 'mushroom', 'broccoli soup', 'tomato curry', 'special roti', 'ice cream', 'veg', 'spice', 'fresh fruits', 'special'}
{'butter', 'roti', 'mutton rogan', 'chicken tikka', 'cheese', 'kulcha', 'lassi', 'chicken', 'malai paneer tikka', 'fish', 'sweet', 'paneer starters', 'naan', 'paneer', 'paneer tikka', 'chicken shorba', 'mixed', 'kabab', 'buttermilk', 'spicy', 'malai paneer', 'milk', 'rabdi', 'lassi shot', 'chicken soup', 'cheese naan', 'dal', 'seekh kabab', 'malai kofta', 'rice', 'onion', 'dal makhani', 'butter chicken', 'kheema', 'jalebi', 'chicken curry', 'amritsari fish', 'spice', 'mutton seekh kabab', 'naan.', 'paratha', 'mutton seekh', 'onion kulcha', 'poi', 'raita'}
{'butter', 'veg starters', 'gulab jamoon', 'sweet', 'noodles', 'chicken', 'paneer', 'paneer tikka', 'naan', 'rasgulla', 'crispy vegetable', 'spicy', 'jamoon', 'onion ring', 'rabdi', 'crispy veg', 'litchi', 'veg', 'onion', 'tea', 'lunch', 'onion rings', 'crispy vegetables', 'veg starter', 'pasta', 'water', 'poi'

{'choco brownie', 'mojito', 'grilled veggies', 'cheese', 'grill', 'lassi', 'chicken', 'brownie', 'egg', 'veg exotic burger', 'choco', 'veggies', 'strawberry bliss mojito', 'brownie shake', 'veggie', 'veg', 'rice', 'chicken patty', 'potato burger', 'hot dog', 'special', 'choco brownie shake', 'strawberry bliss', 'grilled veg', 'potato krunch burger', 'smile', 'tac', 'burger', 'peri wedges', 'wedges', 'strawberry', 'brain', 'fries', 'shakes'}
{'bruschetta', 'sweet', '3 cheese', 'mayo', 'chicken wrap', 'prawns', 'peanut butter', 'cottage cheese', 'coke', 'pizza', 'veggie', 'bread', 'tiramisu', 'rice', 'capsicum', 'cake', 'blueberry shake', 'mushroom', 'bbq pizza', 'butter', 'chai', 'fish', 'chicken', 'regular pizza', 'hot chocolate', 'prawn pizza', 'limited meal', 'spicy', 'choco mocha', 'cheese cake', 'paneer lasagne', 'bacon', 'spice', 'barbeque chicken', 'choco mocha cake', 'wedges', 'pesto prawn pizza', 'veggies', 'fries', 'fresh salad', 'mojito', 'cheese', 'blueberry', 'ida', 'egg', 

{'popcorn', 'chai', 'fish biriyani', 'veg', 'rice', 'fish', 'chicken', 'popcorn chicken', 'poi'}
{'butter', 'pav', 'sweet', 'puri', 'spicy', 'butter masala', 'veg', 'rice', 'dosa', 'uttapam', 'smiles', 'special', 'masala dosa', 'thali', 'smile', 'spice', 'butter masala dosa', 'idly', 'idli', 'churi', 'south indian thali', 'pani puri'}
{'chicken thali', 'roti', 'special chicken thali', 'gobhi paratha', 'special chicken', 'chicken', 'sabji', 'chapati', 'rotis', 'dal fry', 'dal', 'meal', 'keema', 'rice', 'special', 'jamun', 'thali', 'curd', 'tea', 'gulab jamun', 'chicken keema', 'chicken curry', '3 chapati', 'paratha', 'aloo dry', 'poi', 'raita'}
{'thali', 'mushroom', 'malai kofta', 'green peas', 'palak', 'green peas masala', 'mushroom green peas masala', 'paneer', 'peas masala', 'palak paneer'}
{'spice', 'spicy'}
{'fried rice', 'lunch', 'lunch combo', 'rice combo', 'rice', 'churi', 'triple schezwan rice', 'noodles combo', 'noodles', 'schezwan rice', 'manchurian balls', 'fried'}
{'thali',

{'chicken kathi roll', 'kathi roll', 'chicken biriyani', 'chicken noodles.', 'noodles', 'chicken', 'chicken fried rice.', 'egg', 'egg roll', 'roll', 'chicken fried rice', 'chicken 65', 'rice', 'chicken noodles', 'cap', 'fried', 'manchurian chicken', 'chicken egg roll', 'fried rice', 'chicken noodle', 'churi'}
{'coffee'}
{'butter', 'chilli garlic veg pizza', 'veg spicy', 'pizza combo', 'cheese', 'chicken salami pizza', 'double cheese', 'chicken', 'vegetarian pizza', 'cheesy', 'brownie', 'double cheese margherita', 'lava cake', 'regular pizza', 'apple', 'chocolate', 'choco lava', 'salami pizza', 'choco', 'spicy', 'spicy chicken', 'disco', 'chicken pizza', 'spicy chicken pizza', 'french fries', 'cheese margarita pizza', 'bbq mushroom pizza', 'brownie cake', 'om pizza', 'pizza', 'veg pizza', 'coke', 'veg', 'bread', 'meal', 'potato pops', 'rice', 'onion', 'capsicum', 'garlic bread', 'non veg pizza combo', 'chicken salami', 'cake', 'cap', 'veg pizza combo', 'special', 'margarita pizza', 'pin

{'roti', 'grill', 'lollipop', 'chicken', 'paneer chili', 'pop chicken', 'paneer', 'naan', 'roasted', 'chicken starters', 'kebab', 'lollipop chicken', 'parotta', 'spicy', 'kabab', 'rumali', 'tandoori chicken', 'grilled chicken', 'tandoori', 'hariyali kabab', 'shawarma', 'veg', 'rice', 'dragon chicken', 'chicken biryan', 'executive chicken biryani', 'chicken curry', 'lunch', 'tandoori chicken.', 'chicken biryani', 'chicken wings'}
set()
{'khichdi', 'spicy', 'palak khichdi', 'palak', 'sweet', 'papad'}
{'fanta', 'lunch', 'grill', 'chicken biryani', 'chicken', 'grilled chicken', 'chicken biryan', 'special'}
{'jelly', 'tiramisu cake', 'irish', 'irish coffee', 'sweet', 'chocolate', 'choco', 'roll', 'chocolate chip', 'french fries', 'milk', 'coffee', 'smoor sundae', 'tiramisu', 'paan ice cream', 'chocolate based', 'rice', 'cake', 'ice cream', 'dry fruit', 'mango', 'chocolate sauce', 'fries', 'paan', 'packing'}
{'gobi manchurian', 'chicken momos', 'fried chicken', 'lollipop', 'chicken', 'gobi m

{'manchow', 'sweet', 'chicken', 'noodles', 'veg dragon chopsuey', 'paneer', 'ida', 'hot and sour soup', 'exotic vegetables', 'veg hot and sour soup', 'dragon chopsuey', 'exotic vegetable', 'veg', 'meal', 'rice', 'hot and sour', 'manchow soup', 'special', 'pepper salt', 'non-veg', 'mushroom', 'tea', 'chilli paneer', 'lunch', 'mushroom pepper', 'mushroom pepper salt', 'poi'}
{'mayo', 'burger', 'buns', 'french fries', 'fries', 'cutlet', 'bun'}
{'tandoori', 'thali', 'lunch', 'spice', 'carrier meals', 'chicken fry', 'meal', 'tandoori chicken.', 'tandoori chicken', 'chicken', 'chana', 'carrier meal'}
{'poi'}
{'kachori', 'poha', 'palak', 'poi'}
{'chicken rice', 'drums of heaven', 'noodles', 'chicken', 'paneer', 'egg', 'roll', 'garlic chicken', 'garlic noodles', 'hunan chicken sauce', 'hunan chicken', 'veg', 'schezwan veg', 'rice', 'chilli paneer gravy', 'schezwan veg fried rice', 'veg fried rice', 'fried', 'special', 'chilli paneer', 'paneer gravy', 'fried rice', 'dragon rolls', 'schezwan veg

{'classic pizza', 'fanta', 'chai', 'cheese', 'sweet', 'chicken', 'adas', 'lassi', 'vegetarian pizza', 'cheesy', 'sausage', 'apple', 'regular pizza', 'roasted', 'mixed', 'pesto', 'classic', 'pizza margherita', 'roll', 'pepperoni', 'disco', 'lime', 'chicken pizza', 'soft drinks', 'mozzarella pizza', 'packing', 'dal', 'om pizza', 'pizza', 'mozzarella stick', 'coke', 'slice', 'veg', 'bread', 'meal', 'rice', 'onion', 'mozzarella sticks', 'cap', 'cheeseless pizza', 'special', 'margarita pizza', 'bun', 'pineapple', 'mushroom', 'margherita', 'margherita pizza', 'tea', 'make your own', 'lunch', 'toast', 'pasta', 'spice', 'plain cheese', "babe's demise pizza", 'cheese pizza', 'paratha', 'toppings', 'brain', 'spite', 'water', 'poi', 'mushroom pizza'}
set()
{'smile', 'chicken biriyani', 'burger', 'rice', 'lassi', 'chicken', 'milk', 'cake', 'fresh juice', 'sandwich'}
{'sweet', 'meal', 'poori', 'palak'}
{'fried rice', 'water', 'drums of heaven', 'rice', 'dragon chicken', 'chicken', 'noodles', 'churi

{'chicken schezwan noodle', 'noodles', 'chicken', 'chicken schezwan', 'fish', 'ida', 'non veg chilly', 'chicken starters', 'crispy chicken', 'hakka noodles', 'dumpling', 'spicy', 'burnt pepper chicken', 'veg chilly garlic fried rice', 'hakka noodle', 'pepper chicken', 'poi', 'schezwan noodles', 'garlic fried rice', 'veg', 'rice', 'chicken schezwan noodles', 'fried', 'special', 'veg chilly', 'schezwan noodle', 'nam prik chicken', 'fried rice', 'veg chilly garlic', 'lifu chicken', 'chilly garlic fried rice', 'garlic sauce', 'schezwan fried rice', 'packing'}
{'meal', 'pomegranate milk shake', 'rice', 'idli', 'dosa', 'papad', 'milk', '2 idli', 'pomegranate', 'masala dosa'}
{'chicken ghee roast', 'fish', 'sweet', 'chicken', 'clams sukka', 'ghee roast', 'fish curry.', 'prawn ghee roast', 'spicy', 'tawa fry', 'fish meals', 'neer dosa', 'bondas', 'veg', 'fish curry', 'meal', 'rice', 'dosa', 'bondas sukka', 'ghee', 'bonda', 'fried', 'veg curry', 'mackerel', 'fish meal', 'squid sukka'}
{'chocola

{'chicken', 'chicken curry', 'rice', 'spicy'}
{'chicken sandwich', '2 bread', 'bread', 'meal', 'brocolli and almond soup', 'rice', 'bread sandwich', 'chicken', 'almond', 'sandwich'}
{'tea', 'vadapav', 'burger', 'pav', 'fries', 'masala fries', 'vada'}
{'watermelon', 'kulfi', 'banana', 'strawberry', 'water'}
{'roti', 'prawns ghee roast', 'chicken ghee roast', 'fish', 'chicken', 'ghee roast', 'kori roti', 'fish pulimunchi', 'chicken pepper', 'prawns', 'fish meals', 'neer dosa', 'meal', 'rice', 'dosa', 'masala fries', 'ghee', 'pepper dry', 'lunch', 'fish meal', 'kokum', 'fries', 'chicken pepper dry', 'ganji'}
{'tandoori', 'chicken tandoor', 'spicy', 'fish', 'chicken', 'chicken tandoori'}
{'fanta', 'rice'}
{'tea', 'grill', 'pav', 'vada pav', 'egg', 'vada'}
{'aloo gobi', 'hulk', 'phulka'}
{'brownie', 'spicy', 'ice cream', 'fried', 'cheese', 'fried momo', 'chicken', 'noodles', 'paratha', 'fried momos', 'chicken pakora', 'momos', 'egg', 'water'}
set()
{'thali', 'paratha', 'special'}
{'chicken 

{'bhel', 'tikki chat', 'pav', 'pav bhaji', 'sweet', 'dahi puri', 'puri', 'twister', 'ghee jalebi', 'dabeli', 'mixed', 'bhel puri', 'potato twister', 'bhaji', 'aloo tikki', 'paan', 'kachori', 'meal', 'dahi', 'ghee', 'laddu', 'jalebi', 'tea', 'aloo tikki chaat', 'tikki chaat', 'water', 'pani puri'}
{'choco', 'lunch', 'ice cream', 'kulfi', 'choco bar', 'rose', 'falooda', 'lassi', 'sweet', 'kulfi falooda', 'malai kulfi'}
{'thali', 'bread', 'rice', 'grill', 'paratha', 'whole wheat bread', 'maida paratha', 'puri', 'ida', 'spite', 'special', 'kebab'}
{'rajma', 'gulab jamun', 'rajma masala', 'spicy', 'masala paratha', 'paratha', 'paneer', 'latte', 'jamun'}
set()
set()
{'butter', 'roti', 'chicken biriyani', 'chicken 555', 'drums of heaven', 'black pepper', 'chicken tikka', 'chilli mushroom', 'fish', 'chicken', 'paneer chilly', '2 roti', 'chicken kebab', 'puri', 'paneer', 'chicken starters', 'kebab', 'chicken tikka.', 'chicken briyani', 'chicken dum biryani', 'spicy', 'chicken do pyaaza', 'chill

{'butter', 'cheese', 'chicken', 'farm house special pizza', 'cheese steak', 'farm house veg pizza', 'peri peri sauce', 'vada', 'cottage cheese', 'pizza', 'veg pizza', 'khichadi', 'veg', 'african peri peri', 'rice', 'cottage cheese steak', 'special', 'fried', 'tea', 'chicken wings', 'packing'}
{'dal', 'roti', 'dal tadka', 'rotis', 'sabji', 'naan'}
set()
{'waffle', 'waffles', 'tea', 'milk shakes', 'milk', 'fries', 'shakes', 'pasta'}
set()
{'kabab'}
set()
{'undo', 'classic', 'veg', 'burger', 'rice', 'pav', 'lassi', 'classic vada pav', 'vada pav', 'vada'}
{'butter', 'gulkand', 'spicy', 'bun butter', 'bun butter jam', 'masala maggi', 'rice', 'nippat', 'onion', 'panipuri', 'nannari juice', 'puri', 'maggi', 'bun', 'nippat masala', 'packing'}
{'dal', 'jalebi', 'smile', 'kichdi', 'veg', 'dal kichdi', 'sabji', 'poha'}
{'dates', 'apple'}
{'ghee dosa', 'variety rice', 'coffee', 'tea', 'sambar', 'pongal', 'veg', 'lunch', 'poori', 'rice', 'dosa', 'idli', 'poori bhaaji', 'idly', 'idli vada', 'ghee', 

{'fresh salad', 'cheese fries', 'vanilla', 'cupcake', 'stuffed mushroom', 'cheese', 'grill', 'sausages', 'sweet', 'chicken', 'fish', 'cheesy', 'brownie', 'paneer', 'sausage', 'egg', 'chocochip', 'chicken steak', 'cold coffe', 'oreo', 'choco', 'veggies', 'spicy', 'roll', 'cappuccino', 'caramel cappuccino', 'grilled chicken', 'cold coffee', 'cottage cheese sizzler', 'cottage cheese', 'coffee', 'pizza', 'veggie', 'veg', 'bread', 'keema', 'rice', 'hazel', 'onion', 'fish sizzler', 'all meat', 'stuffed mushrooms', 'brownie with vanilla ice cream', 'eggs', 'chicken sizzler', 'stuffed chicken', 'cake', 'cap', 'nachos', 'salad', 'special', 'fish steak', 'mushroom', 'tea', 'special cold coffee', 'all meat pizza', 'dirty fries', 'lunch', 'spice', 'cheesy stuffed mushrooms', 'chicken keema', 'ice cream', 'special coffee', 'wedges', 'vanilla ice cream', 'pepper sauce', 'onion soup', 'hazelnut', 'fries', 'shakes', 'pasta', 'cherry', 'poi'}
{'special chicken', 'sev', 'watermelon', 'grill', 'chicken',

{'manchow', 'chicken manchow soup', 'chicken manchow', 'rice', 'triple schezwan rice', 'soft drinks', 'chicken', 'schezwan rice', 'manchow soup'}
set()
{'mushroom', 'kabab', 'veg', 'rice', 'sweet', 'chicken', 'mushroom dry', 'churi', 'chicken starters', 'sholay kabab'}
{'mojito', 'black current milkshake', 'black current shake', 'rocher', 'fish', 'noodles', 'chicken', 'fries', 'pasta salad', 'egg', 'cold coffe', 'special fried rice', 'mint mojito', 'black current', 'milk', 'aqua', 'cold coffee', 'veggie', 'coffee', 'veg', 'rice', 'red rice', 'milkshake', 'chicken sizzler', 'salad', 'special', 'fried', 'tea', 'fried rice', 'white sauce pasta', 'spice', 'veggies', 'shakes', 'pasta', 'poi'}
{'dal', 'cheese', 'cheese balls', 'cheese ball', 'milk', 'milkshake', 'shakes', 'ida', 'hide and seek', 'sandwich'}
set()
{'special'}
set()
{'roti', 'spicy', 'meal', 'rotis', 'papad', 'sabji'}
{'last night'}
{'death by chocolate', 'choco', 'cheesecake', 'red velvet', 'cheese', 'custard', 'chocolate cak

{'roll', 'lemon chicken', 'drums of heaven', 'rice', 'chicken'}
set()
{'butter', 'paneer 65', 'manchow', 'roti', 'chicken biriyani', 'cupcake', 'kori rotti', 'chicken manchow soup', 'anise', 'pepper fry', 'cream of mushroom soup', 'churi', 'baby corn chilly', 'prawns ghee roast', 'fish', 'chicken', 'fries', 'paneer', 'naan', 'latte', 'egg', 'squid chilly', 'ghee roast', 'oreo', 'spicy', 'buttermilk', 'combo meal', 'rotis', 'rasam', 'prawns', 'milk', 'pappad', 'neer dosa', 'baby corn', 'chicken sukka', 'veggie', 'malai kofta', 'veg', 'crab', 'bread', 'chicken 65', 'meal', 'rice', 'dosa', 'stress buster', 'chapatti', 'roti.', 'cake', 'manchow soup', 'prawns masala', 'ghee', 'special', 'butter chicken', 'fried', 'pomfret fry', 'cream of mushroom', 'tac', 'mushroom', 'curd', 'thali', 'veg biryani', 'lunch', 'spice', 'veg curry', 'chicken curry', 'chicken manchow', 'mangalore chicken curry', 'fried rice', 'baby corn chil', 'fish meal', 'mushroom soup', 'veggies', 'cherry', 'pomfret', 'poi'}

{'tea', 'full meal', 'lunch', 'spice', 'meal', 'chicken biryani', 'boneless chicken biryani', 'chicken', 'chicken biryan', 'boneless chicken', 'peni', 'water'}
{'roti', 'hulk', 'bhurji', 'meal', 'rotis', 'rice', 'bhatura', 'ajwain phulka', 'phulka', 'sweet', 'potato stew', 'paneer bhurji', 'peni', 'paneer', 'poi', 'packing'}
{'veg', 'meal', 'chilly paneer', 'paratha', 'paneer', 'veg meal'}
{'rice', 'sweet', 'lime', 'fudge'}
set()
{'fanta', 'croissant', 'pastry', 'bread', 'sweet', 'cake', 'spite', 'bun'}
{'chai', 'kulfi', 'badam kulfi', 'sweet', 'rainbow', 'lassi', 'apple', 'chocolate', 'hot chocolate', 'custard apple', 'rabdi kulfi', 'choco', 'roll', 'custard', 'chocolate kulfi', 'milk', 'alphonso mango', 'rabdi', 'rabri', 'strawberry', 'malai kulfi', 'hot choco', 'dry fruit kulfi', 'rice', 'kesar badam kulfi', 'special', 'choco almond', 'strawberry lassi', 'tea', 'kesar badam', 'badam', 'ice cream', 'dry fruit', 'pista kulfi', 'falooda', 'almond', 'kulfi ice cream', 'mango', 'pista', 

{'choco', 'ice cream', 'potato twister', 'rice', 'chocolate sauce', 'twister', 'chocolate'}
{'sambar', 'tea', 'chicken biriyani', 'boneless chicken biriyani', 'spicy', 'veg', 'chicken', 'chicken dum biriyani', 'boneless chicken', 'chilli chicken', 'poi'}
{'butter', 'vanilla', 'chai', 'butter scotch', 'red velvet', 'cheese', 'red velvet cake', 'chocolate', 'oreo', 'choco', 'disco', 'chocolate ice cream', 'poi', 'litchi', 'cheesecake', 'red velvet ice cream', 'meal', 'butterscotch', 'rice', 'blue sparkle', 'cake', 'salted caramel', 'special', 'waffle', 'smile', 'nutty professor sundae', 'ice cream', 'almond praline', 'thumbs up', 'almond', 'fudge', 'toppings', 'waffle cone', 'packing'}
{'tandoori', 'roti', 'poi', 'malai chicken', 'meal', 'rotis', 'rice', 'chicken', 'special', 'tandoori roti'}
{'non-veg', 'spicy', 'veg', 'biryani rice', 'rice', 'bun', 'chapati', 'mutton chops dry', 'poi', 'mutton chops'}
{'rice', 'milk', 'milkshake', 'shakes', 'sandwich'}
set()
{'vanilla', 'red velvet', '

{'vadapav', 'samosa', 'peda', 'pav', 'vada pav', 'water', 'vada'}
set()
{'spicy', 'roll', 'rice', 'grill', 'chicken'}
{'veg', 'special', 'kabab'}
set()
set()
set()
{'coffee', 'filter coffee', 'idli'}
{'ice tea', 'pizza', 'tea', 'indian pizza', 'rice', 'pasta'}
{'spice', 'sev', 'chicken tikka', 'basmati rice', 'rice', 'family pack biryani', 'chicken boneless biryani', 'kpl', 'chicken', 'rabadi', 'jamun', 'boneless biryani'}
{'butter', 'roti', 'plain paratha', 'paneer butter', 'tomato rice', 'paneer chilly', 'paneer', 'rotis', 'paneer butter masala', 'butter masala', 'dal', 'meal', 'rice', 'salad', 'special', 'fried', 'fried rice', 'paratha', 'churi', 'paneer paratha', 'spite', 'water', 'toppings'}
{'rice', 'fish and chips', 'fish', 'special', 'fried'}
set()
{'spice', 'spicy', 'egg', 'egg biryani'}
set()
set()
{'chole', 'roti', 'kadhai paneer', 'spice', 'rice', 'soya chaap', 'chole masala', 'roti.', 'paneer'}
{'kabab'}
set()
{'chicken meals', 'roti', 'chicken meal', 'chicken curry', 'spe

{'water', 'lime ginger', 'lime', 'rice', 'sugarcane juice', 'special'}
{'keema vada pav', 'non-veg', 'samosa', 'veg', 'burger', 'ice cream', 'momos', 'spicy', 'keema', 'rice', 'veg momos', 'pav', 'vada', 'supreme', 'vada pav', 'veg momo', 'veg burger', 'sprouts'}
set()
{'rice'}
set()
set()
set()
set()
set()
set()
set()
{'veg & non-veg', 'non-veg', 'veg', 'rice'}
set()
set()
set()
{'butter', 'roti', 'paneer butter', '2 roti', 'hara bhara kebab', 'puri', 'paneer tikka', 'pulav', 'paneer', 'kebab', 'rotis', 'papad', 'paneer butter masala', 'butter masala', 'sandwich', 'dhokla', 'dal', 'rice', 'rice combo', 'jamun', 'thali', 'masala puri', 'gulab jamun', 'jeera rice', 'poi', 'raita'}
{'dal', 'lunch', 'fish curry', 'rice', 'fish'}
set()
set()
set()
set()
set()
set()
set()
set()
set()
{'bruschetta', 'fanta', 'chai', 'veg starters', 'cheese', 'grill', 'chicken ghee roast', 'cheese ball', 'fish', 'chicken', 'tomato bruschetta', 'cock', 'chicken kebab', 'coastal chicken', 'ghee roast', 'drumsti

{'undo', 'roti', 'chai', 'kulfi', 'bombay', 'halwa', 'masala papad', 'chilli mushroom fried rice', 'chilli mushroom', 'sweet', 'noodles', 'apple', 'paneer', 'puri', 'ida', 'naan', 'egg', 'brownie', 'chole', 'spicy', 'kabab', 'rotis', 'mushroom fried rice', 'papad', 'rabri', 'veg kolapuri', 'water', 'ice tea', 'rajma', 'garlic naan', 'veg lovers', 'seekh kabab', 'veg', 'bread', 'rice', 'roti.', 'hot and sour', 'special', 'fried', 'sizzling brownie', 'bun', 'mushroom', 'tea', 'fried rice', 'lunch', 'ice cream', 'spice', 'veg biryani', 'churi', 'fries', 'pasta', 'paan', 'poi', 'raita'}
set()
{'spice', 'mutton kebab', 'dahi', 'fish', 'fish tikka', 'dahi kebab', 'kebab'}
{'tuna', 'lunch'}
{'special tea', 'bruschetta', 'banana waffle', 'sev', 'grill', 'mix sauce', 'pav', 'nutella', 'sweet', 'lassi', 'earl grey tea', 'latte', 'pav bhaji fondue', 'classic', 'masala chai', 'mayo', 'milk', 'milk chocolate', 'brownie shake', 'earl grey', 'ice tea', 'cottage cheese', 'veggie', 'coffee', 'lime soda

{'cheese', 'cold brew', 'blueberry', 'chocolate', 'hot chocolate', 'cappuccino', 'choco', 'cheese cake', 'milk', 'berry delight', 'blueberry muffin', 'green tea', 'hot choco', 'frappuccino', 'coffee', 'meal', 'cake', 'cap', 'special', 'espresso', 'tea', 'lunch', 'mocha'}
{'coffee', 'veg samosa', 'samosa', 'caramel coffee', 'veg', 'cheesecake', 'cheese', 'thumbs up', 'chicken', 'cheese cake', 'cake', 'chicken samosa'}
set()
{'butter', 'chai', 'ice coffee', 'paneer sandwich', 'cheese', 'lemon tea', 'lassi', 'sweet', 'paneer', 'chocolate', 'hot chocolate', 'chocolate sauce', 'cold coffe', 'cappuccino', 'choco', 'mayo', 'spicy', 'plain coffee', 'hazelnut cappuccino', 'disco', 'cheese corn balls', 'potato wedges', 'chocolate milk', 'french fries', 'milk', 'cheese corn ball', 'sandwich', 'cold coffee', 'hot choco', 'cheese corn', 'peanut butter', 'coffee', 'pizza', 'mayonnaise', 'veg', 'peanut butter shake', 'bread', 'hazel', 'potato fries', 'rice', 'cake', 'ginger lemon tea', 'cap', 'cookie

{'lasagne', 'sev', 'bombay', 'cheese', 'basmati rice', 'chicken', 'cheesy', 'egg', 'chocolate', 'choco', 'spicy', 'bombay masala quesadilla', 'veggie', 'chicken lasagna', 'lasagna', 'minced chicken', 'veg', 'bread', 'cheesecake', 'rice', 'cake', 'tac', 'chocolate decadence', 'spice', 'veggies', 'pasta', 'water'}
{'butter', 'chicken rolls', 'chai', 'samosa', 'butter bun', 'fried chicken', 'lemon tea', 'pav', 'chicken', 'veg roll', 'sharbath', 'cutlet', 'mixed', 'masala chai', 'roll', 'masala c', 'kabab', 'chicken keema roll', 'custard', 'double chicken roll', 'milk', 'keema roll', 'vada', 'lemon juice', 'kadi', 'veg', 'masala milk', 'milk tea', 'chicken roll', 'rice', 'keema', 'chicken cutlet', 'cutlets', 'mixed roll', 'rabadi', 'caramel custard', 'vada pav', 'fried', 'special', 'bun', 'tea', 'chicken keema', 'regular tea', 'burger', 'hyderabadi chicken', 'sharbat', 'water'}
{'butter', 'jelly', 'grape', 'grill', 'rocher', 'lassi', 'chocolate', 'choco', 'falooda ice cream', 'lime', 'gril

{'chicken noodle', 'rice', 'onion', 'chicken noodles', 'chicken', 'noodles', 'churi', 'red chilly', 'chilly chicken'}
{'chicken rolls', 'roti', 'mutton sheek', 'grill', 'beef chilly', 'sweet', 'chicken', 'fish', 'kebab', 'mayo', 'roll', 'grilled chicken', 'kalmi kebab', 'shawarma', 'mutton seek roll', 'mayonnaise', 'chicken roll', 'chicken shawarma', 'rice', 'beef shawarma', 'special', 'kalmi', 'spice', 'burger', 'shawarma roll', 'strawberry', 'water', 'poi'}
{'corn roll', 'spicy', 'roll', 'grill', 'rice', 'chicken biryani', 'tikka roll', 'grilled baby corn roll', 'chicken', 'grilled baby corn', 'chicken biryan', 'baby corn roll', 'baby corn'}
{'aloo gobi', 'veg cheese sandwich', 'cheese', 'lassi', 'sweet', 'aloo paratha', 'gobi paratha', 'cheese sandwich', 'mixed', 'oreo', 'mayo', 'roll', 'milk', 'water', 'sandwich', 'mixed sauce pasta', 'sweet lassi', 'aloo gobi paratha', 'paratha.', 'veg', 'onion', 'milkshake', 'pickle', 'nachos', 'fried', 'special', 'curd', 'white sauce pasta', 'bu

{'gulkand', 'gulkand kulfi', 'kesar badam', 'badam', 'sev', 'kulfi', 'badam kulfi', 'rice', 'coconut kulfi', 'kesar badam kulfi', 'tender coconut', 'special'}
{'sev', 'panner tikka', 'lollipop', 'noodles', 'chicken', 'fish', 'apple', 'chicken lollypop', 'latte', 'momos', 'veg roll', 'kabab', 'roll', 'lime', 'chicken lollipop', 'lollypop', 'pizza', 'veg', 'rice', 'special', 'mushroom', 'spice', 'last night', 'babycorn manchurian', 'churi', 'fries'}
{'khichdi', 'chicken manchow soup', 'grill', 'sweet', 'kabab', 'chicken gravy', 'rotis', 'tandoori chicken', 'grilled chicken', 'prawns', 'dal', 'veggie', 'shawarma', 'bread', 'rice', 'mutton gravy', 'non-veg', 'mushroom', 'tac', 'afgani chicken', 'chicken hydrabadi', 'butter', 'roti', 'fanta', 'butter roti', 'fish', 'chicken', 'lollipop', 'paneer tikka', 'dragon mushroom', 'momos', 'kebab', 'spicy', 'roll', 'tikka biryani', 'egg biryani', 'rumali', 'chilli chicken', 'veg lovers', 'manchow soup', 'ghee', 'hyderabadi biryani', 'lunch', 'spice'

{'roti', 'khichdi', 'sweet curd', 'pav', 'pav bhaji', 'sweet', 'lassi', 'shrikhand', 'dabeli', 'mixed', 'sabudana vada', 'vada', 'bhaji', 'pickle', 'vada pav', 'poha', 'peanut', 'thali', 'curd', 'sabudana'}
{'coffee', 'filter coffee', 'rice'}
set()
{'waffle', 'waffles', 'oreo', 'fried'}
{'roti', 'samosa', 'latte', 'pasta salad', 'ida', 'paneer', 'cream of tomato soup', 'hulk', 'papad', 'steamed rice', 'dal', 'veg', '2 phulka', 'bread', 'meal', 'rice', 'cream of tomato', 'tomato soup', 'capsicum', 'phulka', 'cap', 'salad', 'tac', 'tea', 'curd', 'lunch', 'babycorn manchurian', 'churi', 'pasta'}
set()
{'chicken momo', 'tandoori', 'garlic naan', 'chicken momos', 'spicy', 'kulfi', 'fish', 'chicken', 'naan', 'momos', 'chicken wings'}
{'paratha.', 'lunch', 'rice', 'kulcha', 'paratha', 'paneer', 'latte', 'water', 'lacha paratha'}
{'chilli guava', 'drums of heaven', 'black pepper', 'fish', 'chicken', 'noodles', 'jasmine tea', 'lollipop', 'paneer', 'momos', 'capsicum noodles', 'chocolate', 'hot 

{'soda', 'chai', 'watermelon', 'chicken', 'veg burger', 'mayo', 'spicy', 'milk', 'masala soda', 'chicken spicy burger', 'mayonnaise', 'veg', 'rice', 'milkshake', 'bun', 'mushroom', 'chicken burger', 'burger', 'spice', 'fries', 'water', 'watermelon masala'}
{'cupcake', 'brownies', 'eggless', 'cake', 'brownie', 'cookie', 'macarons', 'egg'}
{'sambar', 'samosa', 'noodles', 'vada sambar', 'mushroom manchuria', 'set dosa', 'mushroom manchuri', 'soft drinks', 'vada', 'mushroom manchurian', 'veg', 'rice', 'dosa', 'onion', 'fried', 'idly vada', 'mushroom', 'onion dosa', 'fried rice', 'idly', 'churi'}
{'butter', 'kesari', 'choco', 'kesaribath', 'bread', 'lime', 'omelette', 'upma', 'egg', 'vada', 'poi', 'omelet'}
{'orange', 'classic', 'lassi', 'smoothie', 'milk', 'milkshake', 'shakes', 'pomegranate', 'pomegranate juice'}
set()
{'butter', 'badam milk', 'butter cookies', 'coffee', 'tea', 'filter coffee', 'badam', 'coffee shakes', 'coffee shake', 'rice', 'cold beverage', 'milk', 'normal tea', 'cold 

{'butter', 'blueberry cake', 'white forest', 'butter scotch', 'cheese', 'sweet', 'blueberry', 'egg', 'chocolate', 'choco', 'rusk', 'cheese cake', 'peni', 'pizza', 'mini pizza', 'rice', 'cake', 'cookie', 'pastry', 'chocolate cake', 'eggless', 'strawberry', 'white forest pastry', 'water'}
set()
set()
set()
{'chicken sandwich', 'mix veg', 'bun omlette', 'omlette', 'veg', 'bread', 'cheese', 'grill', 'chicken', 'veg cheese grill', 'sandwich', 'bun'}
{'apple cake', 'butter', 'veg puffs', 'fruit cake', 'legendary', 'veg', 'bread', 'veg puff', 'honey cake', 'rice', 'buns', 'congress', 'cake', 'apple', 'potato bun', 'groundnut', 'special', 'bun'}
{'hot chocolate fudge', 'apple', 'chocolate', 'hot chocolate', 'almond fudge', 'chocolate sauce', 'choco', 'black forest', 'apple crumble', 'evergreen', 'hot choco', 'death by chocolate', 'black forest cake', 'rice', 'cake', 'groundnut', 'cashew', 'chocolate fudge', 'ice cream', 'dry fruit', 'almond', 'fudge', 'cake fudge', 'spite', 'water', 'poi', 'dr

{'choco lava', 'pizza', 'veg pizza', 'choco', 'veg', 'farmhouse', 'cheese', 'veg farmhouse pizza', 'choco lava cake', 'capsicum', 'cake', 'cap', 'lava cake', 'special', 'farmhouse pizza', 'poi'}
{'coffee', 'tea', 'idly', 'rice', 'vada'}
{'cheese', 'ida', 'chole', 'spicy', 'masala lemon', 'lemonade', 'pizza', 'veg pizza', 'slice', 'veg', 'bread', 'meal', 'rice', 'onion', 'garlic bread', 'chole masala', 'margherita', 'tea', 'spice', 'overload', 'toppings', 'masala lemonade'}
{'prawns sukka', 'prawns ghee roast', 'chicken ghee roast', 'fish', 'chicken', 'ghee roast', 'kabab', 'bangda', 'prawns', 'tawa fry', 'fish meals', 'chicken sukka', 'dal', 'bangda tawa fry', 'meal', 'fish curry', 'rice', 'dosa', 'ghee', 'special', 'tea', 'bangda tawa', 'spice', 'fish meal', 'dosa chicken'}
{'roti', 'slice', 'spicy', 'fried rice', 'roll', 'special', 'rice', 'onion', 'noodles', 'churi', 'fried'}
{'hot beverage', 'sambar', 'single idli vada', 'sev', 'single idli', 'poori sagu', 'halwa', 'sweet', 'adas',

{'dal', 'roti', 'thali', 'fried rice', 'sev', 'hulk', 'jeera rice', 'rice', 'lassi', 'paratha', 'phulka', 'special', 'fried'}
{'rice'}
{'rice'}
set()
set()
{'roti', 'chai', 'masala dosa', 'meal', 'south meal', 'rice', 'dosa', 'chapathi', 'special', 'poi', 'parotta'}
set()
set()
{'2 parota', 'cop', 'parota', 'roll', 'chicken roll', 'chicken', 'special'}
set()
{'meal', 'special', 'pickle'}
set()
set()
{'butter', 'sambar', 'akki rotti', 'veg south indian', 'sweet', 'bele bath', 'lassi', 'rice bath', 'pulav', 'classic', 'rava dosa', 'milk', 'butter milk', 'vada', 'green tea', 'coffee', 'veg', 'meal', 'rice', 'dosa', 'ghee', 'special', 'masala dosa', 'medu vada', 'bisibele bath', 'tea', 'lunch', 'pongal', 'idly', 'idli', 'poi'}
{'sambar', 'fanta', 'veg starters', 'halwa', 'nutella', 'sweet', 'chicken', 'fish', 'brownie', 'ground nut', 'naan', 'paneer', 'veg roll', 'chocolate', 'salads', 'choco', 'kheer', 'roll', 'gajar halwa', 'jamoon', 'papdi chaat', 'steamed rice', 'brownie cake', 'dal', 

{'crab curry', 'grill', 'sweet', 'firni', 'corn kernels', 'corn kernel', 'chana', 'prawns', 'ice tea', 'pan ice cream', 'dal', 'veggie', 'tangdi kebab', 'crab', 'paan ice cream', 'rice', 'sarson da saag', 'cake', 'mushroom', 'lamb seekh kebab', 'luchi', 'ice cream', 'grilled prawns', 'roti', 'watermelon', 'fish', 'chicken', 'paneer tikka', 'rasgulla', 'kebab', 'spicy', 'roll', 'spirit', 'makai roti', 'fish curry', 'lunch', 'spice', 'chana dal', 'veggies', 'halwa', 'apple', 'egg', 'white pasta', 'makai ki roti', 'steamed rice', 'gilafi seekh', 'veg', 'mixed fruit', 'rabadi', 'special', 'fried', 'jamun', 'chicken curry', 'churi', 'spite', 'paan', 'poi', 'mustard fish', 'kofta curry', 'paneer', 'sandesh', 'mixed', 'crispy corn', 'disco', 'peni', 'paneer pasanda', 'meal', 'palak', 'eggs', 'misti doi', 'bhaja', 'paneer pasand', 'salad', 'murgh tikka', 'pineapple', 'tea', 'fried rice', 'banana', 'aloo posta', 'pasta', 'water'}
{'boiled veg', 'pakoda', 'grill', 'fish wrap', 'lassi', 'sweet', 

{'chai', 'falafel bowl', 'grill', 'sweet', 'chicken', 'fruit bowl', 'detox water', 'latte', 'salads', 'roll', 'stuffed omelette', 'milk', 'sandwich', 'ginger chai', 'mop', 'shawarma', 'veg', 'ginger tea', 'rice', 'chicken shawarma', 'salad', 'chicken bowl', 'chicken grilled burger', 'tea', 'falafel', 'lunch', 'burger', 'detox', 'omelette', 'chicken grilled', 'spite', 'water', 'omelet'}
{'coffee', 'caramel latte', 'chai', 'rice', 'lassi', 'milk', 'latte', 'water'}
{'cold coffe', 'coffee', 'choco', 'croissant', 'chocolate donut', 'almond', 'cake', 'cup cakes', 'club sandwich', 'donut', 'chocolate', 'sandwich', 'cold coffee'}
{'coffee', 'tea', 'chai', 'roll', 'iced tea'}
{'cheese', 'double cheese', 'chicken', 'mint green tea', 'puri', 'latte', 'egg', 'chocolate', 'cold coffe', 'coffee lover', 'cappuccino', 'choco', 'crunchy veg', 'lime', 'chilly cheese toast', 'lime and mint', 'frappe', 'cold coffee', 'sandwich', 'green tea', 'coffee', 'pizza', 'choco shot', 'veggie', 'veg', 'chilly chees

{'cheese', 'grill', 'mac and cheese', 'chicken', 'noodles', 'cheesy', 'cheesy chicken popcorn', 'egg', 'chicken pop', 'chicken steak', 'popcorn', 'mixed', 'spicy', 'masala c', 'grilled chicken', 'cheesy chicken', 'veggie', 'veg', 'lays', 'grilled chicken steak', 'rice', 'cake', 'chicken biryan', 'chicken popcorn', 'bun', 'tea', 'dry fruit', 'potato chips', 'chicken biryani', 'veggies', 'masala chicken'}
{'butter', 'butter naan', 'kadhai paneer', 'sweet', 'lassi', 'adas', 'papdi chat', 'naan', 'paneer', 'aloo curry', 'upma', 'vada', 'coffee', 'sweet lassi', 'meal', 'poori', 'rice', 'dosa', 'shahi paneer', 'papdi', 'ghee', 'salad', 'sambhar', 'thali', 'idli', 'churi', 'shakes', 'south indian thali'}
set()
{'mixed', 'lemon pickle', 'spicy', 'lunch', 'appam', 'fish curry', 'rice', 'achari chicken', 'natholi', 'fish', 'prawn masala', 'chicken', 'pickle', 'special', 'natholi fry'}
{'butter', 'fanta', 'aloo puri', 'full meal', 'cheese', 'sweet', 'aloo paratha', 'puri', 'paneer', 'dhaniya chut

{'butter', 'roti', 'butter naan', 'sev', 'chicken tikka', 'green peas', 'grill', 'grilled fish', 'fish', 'chicken', 'naan', 'pepper chicken masala', 'kebab', 'parota', 'chicken masala', 'mayo', 'spicy', 'kabab', 'tikka biryani', 'chicken gravy', 'rotis', 'grilled chicken', 'pepper chicken', 'chicken tikka biryani', 'veg lovers', 'chicken hyderabadi', 'veg', 'bread', 'rice', 'rabadi', 'toum', 'ghee', 'chicken hyderabad', 'hyderabadi biryani', 'ghee rice', 'non-veg', 'chicken hyderabadi biryani', 'lunch', 'grill chicken', 'hummus', 'churi', 'poi'}
{'butter', 'special chicken biryani', 'special chicken', 'sev', 'paneer butter', 'paneer tikka biryani', 'fish', 'chicken', 'chicken kebab', 'boiled egg', 'naan', 'paneer tikka', 'bamboo chicken', 'chicken starters', 'egg', 'brinjal biryani', 'kebab', 'paneer', 'tangdi kabab', 'veg & veg', 'ida', 'chicken crispy', 'veg thali', 'spicy', 'kabab', 'hulk', 'tikka biryani', 'lime', 'kpl', 'boiled egg.', 'garlic chicken', 'paneer butter masala', 'but

{'butter', 'butter naan', 'tea', 'chai', 'lunch', 'spice', 'veg', 'paneer masala', 'palak', 'naan.', 'butter paneer', 'naan', 'rice', 'paneer', 'special', 'butter paneer masala', 'butter naan.', 'palak paneer'}
set()
{'kabab', 'biryani rice', 'meal', 'rice', 'chicken', 'chilly chicken'}
{'hyderabadi mutton biryani', 'non veg combo', 'seu', 'sev', 'chicken tikka', 'kesari phirni', 'sweet', 'chicken', 'claypot hyderabadi chicken biryani', 'paneer tikka', 'paneer', 'hyderbadi biryani', 'ida', 'kesari', 'mixed', 'tikka biryani', 'packing', 'hyderabadi mutton', 'chicken tikka biryani', 'veg combo', 'boneless chicken tikka', 'veg', 'mutton biryani', 'meal', 'rice', 'chicken biryan', 'rabadi', 'special', 'jamun', 'mix veg', 'non-veg', 'gulab jamun', 'veg biryani', 'lunch', 'spice', 'hyderabadi chicken biryani', 'chicken biryani', 'hyderabadi chicken', 'paneer biryani', 'boneless chicken', 'mutton biriyani', 'phirni', 'poi', 'raita'}
{'butter', 'roti', 'butter naan', 'chicken biriyani', 'chick

{'butter', 'lasagne', 'chicken mac n cheese', 'bombay', 'chicken tikka', 'tandoori kukkad', 'cheese', 'paneer steak', 'chicken', 'cheesy', 'desi videsi', 'paneer', 'roasted', 'egg', 'chocolate', 'choco', 'mac n cheese', 'chocolate lover', 'bombay masala quesadilla', 'tandoori', 'veggie', 'chicken lasagna', 'lasagna', 'butter chicken mac n cheese', 'veg', 'minced chicken', 'meal', 'rice', 'butter chicken', 'tea', 'lunch', 'veggies', 'pasta', 'packing'}
{'butter', 'butter naan', 'chicken biriyani', 'chai', 'sweet', 'chicken', 'naan', 'roasted', 'spicy', 'tandoori chicken', 'mushroom masala', 'chilli chicken', 'tandoori', 'shawarma', 'veg lovers', 'veg', 'rice', 'special', 'mushroom', 'chicken curry', 'lunch', 'spice', 'mast', 'spicy shawarma', 'poi'}
{'mazza'}
{'roti', 'fanta', 'full meal', 'khichdi', 'paneer makhani', '2 bread', 'chicken tikka', 'pakoda', 'cheese', 'aloo paratha meal', 'sausages', 'sweet', 'chicken', 'aloo paratha', 'capsicum curry', 'sabji', 'cheesy', 'paneer', 'paneer

{'choco', 'cupcake', 'rocher', 'cake', 'chocolate'}
{'veggie', 'veg', 'noodles', 'veggies', 'egg'}
{'manchow', 'chicken manchow soup', 'kulcha', 'chicken', 'chilli prawns', 'kebab', 'chicken briyani', 'mixed', 'spicy', 'boneless chicken biryani', 'red hot chicken', 'prawns', 'baby corn', 'pepper chicken', 'chilli prawn', 'veg', 'rice', 'chicken biryan', 'rabadi', 'manchow soup', 'special', 'veg crispy', 'thali', 'lunch', 'family biryani', 'chicken manchow', 'chicken biryani', 'boneless chicken', 'poi'}
{'cool breeze', 'roti', 'baklava', 'chai', 'cheese spring roll', 'chicken tikka', 'cheese', 'fish', 'chicken', 'lassi', 'cock', 'sweet', 'sausage', 'classic chicken tikka', 'kebab', 'cutlet', 'popcorn', 'chicken wings', 'classic', 'roll', 'chicken pizza', 'fish tikka', 'peni', 'aqua', 'spinach and corn quiche', 'cherry', 'chilli chicken', 'pizza', 'chicken sausage', 'chicken cutlet', 'rice', 'roti.', 'cake', 'chilli cheese', 'ajwaini fish tikka', 'special', 'chicken bao', 'spring rolls',

AttributeError: 'NoneType' object has no attribute 'replace'

After analysing the above output these were some of the frequent non menu words found.

In [16]:
remove_words = ['poi', 'spite', 'churi', 'peni', 'packing', 'meal', 'malt', 'smilies', 'disco', 'tac', 'cap', 'spicy', 'last night', 'cop', 
'brain', 'onion', 'spice', 'holige', 'adas', 'sulaimani', 'special', 'combo 4', 'toppings', 'smile', 'water', 'ghee', '',
'mixed', 'slice', 'lunch', 'masala c', 'churi', 'breakfast combo', 'undo', 'full meal', 'fanta', 'cool blue', 'love', 'veg lovers',
'irish', 'fried', 'mast', 'rose', 'overload']
# Lets remove them from our corpus first
new_menu_corpus = [word for word in menu_corpus if word not in remove_words]
len(new_menu_corpus)

71363

In [17]:
# perfect now those noise are removed, let us extract menus from review and store it in the column liked_food_from_review

df['liked_food_from_review'] = df['liked_food_from_review'].astype(object)
df['liked_food_from_review'] = "[]"

KeyError: 'liked_food_from_review'